hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -295,6 +311,11 @@ protected HoodieMetadataPayload(String key, int type,
     this.recordIndexMetadata = recordIndexMetadata;
   }
 
+  public HoodieMetadataPayload(String key, HoodieSecondaryIndexInfo secondaryIndexMetadata) {
```

Comment:
should it be public scope or private, move it to line 301 seems better.

Reply:
Should be private. Refactored as suggested.